### ISSUE TITLE
kata v2.3/v2.4 doesn't support the configuration `enable_swap = true`

### ISSUE THREAD

**Issue by Bevisy** [kata-containers/kata-containers#3221]
# Description of problem
When setting `enable_swap = true`, pod will create failed

# Expected result
pod failed

# Actual result
pod running

# Further information
error log as below:
```sh
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.626646664+08:00" level=info name=containerd-shim-v2 path=/opt/kata/libexec/kata-qemu/virtiofsd pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=virtiofsd
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.626666028+08:00" level=info args="--syslog -o cache=auto -o no_posix_lock -o source=/run/kata-containers/shared/sandboxes/57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464/shared --fd=3 -f --thread-pool-size=1 -o log_level=debug" name=containerd-shim-v2 pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=virtiofsd
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.626884774+08:00" level=info msg="Adding extra file [0xc000128140 0xc000128230 0xc000128210]" name=containerd-shim-v2 pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=qmp
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.626936095+08:00" level=info msg="launching /opt/kata/bin/qemu-system-x86_64 with: [-name sandbox-57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 -uuid 5f017240-0eb4-49fd-81c1-fc6a7c889659 -machine q35,accel=kvm,kernel_irqchip=on -cpu host,pmu=off -qmp unix:/run/vc/vm/57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464/qmp.sock,server=on,wait=off -m 2048M,slots=10,maxmem=16852M -device pci-bridge,bus=pcie.0,id=pci-bridge-0,chassis_nr=1,shpc=off,addr=2,io-reserve=4k,mem-reserve=1m,pref64-reserve=1m -device virtio-serial-pci,disable-modern=false,id=serial0 -device virtconsole,chardev=charconsole0,id=console0 -chardev socket,id=charconsole0,path=/run/vc/vm/57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464/console.sock,server=on,wait=off -device virtio-scsi-pci,id=scsi0,disable-modern=false -object rng-random,id=rng0,filename=/dev/urandom -device virtio-rng-pci,rng=rng0 -device pcie-root-port,id=rp0,bus=pcie.0,chassis=0,slot=0,multifunction=off -device pcie-root-port,id=rp1,bus=pcie.0,chassis=0,slot=1,multifunction=off -device vhost-vsock-pci,disable-modern=false,vhostfd=3,id=vsock-3904803025,guest-cid=3904803025 -chardev socket,id=char-2b1fc97a054bb5fe,path=/run/vc/vm/57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464/vhost-fs.sock -device vhost-user-fs-pci,chardev=char-2b1fc97a054bb5fe,tag=kataShared -netdev tap,id=network-0,vhost=on,vhostfds=4,fds=5 -device driver=virtio-net-pci,netdev=network-0,mac=56:36:d9:a1:f7:a9,disable-modern=false,mq=on,vectors=4 -rtc base=utc,driftfix=slew,clock=host -global kvm-pit.lost_tick_policy=discard -vga none -no-user-config -nodefaults -nographic --no-reboot -daemonize -object memory-backend-file,id=dimm1,size=2048M,mem-path=/dev/shm,share=on -numa node,memdev=dimm1 -realtime mlock=off -kernel /opt/kata/share/kata-containers/vmlinux-5.10.25-87 -initrd /opt/kata/share/kata-containers/kata-alpine-3.13.5.initrd -append tsc=reliable no_timer_check rcupdate.rcu_expedited=1 i8042.direct=1 i8042.dumbkbd=1 i8042.nopnp=1 i8042.noaux=1 noreplace-smp reboot=k console=hvc0 console=hvc1 cryptomgr.notests net.ifnames=0 pci=lastbus=0 quiet panic=1 nr_cpus=8 scsi_mod.scan=none agent.log=debug agent.debug_console agent.debug_console_vport=1026 -pidfile /run/vc/vm/57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464/pid -smp 1,cores=1,threads=1,sockets=8,maxcpus=8]" name=containerd-shim-v2 pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=qmp
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.629378779+08:00" level=error msg="Unable to launch /opt/kata/bin/qemu-system-x86_64: exit status 1" name=containerd-shim-v2 pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=qmp
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.629404483+08:00" level=error msg="qemu-system-x86_64: -realtime: invalid option\n" name=containerd-shim-v2 pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=qmp
12月 07 18:43:54 zbb-pc kata[755701]: time="2021-12-07T18:43:54.629424969+08:00" level=error msg="failed to launch qemu: qemu-system-x86_64: -realtime: invalid option\n" error="exit status 1" name=containerd-shim-v2 pid=755701 sandbox=57791640635be1e583d012deccd6e5deaf727a3ae3ada5f047d7dd735869d464 source=virtcontainers/hypervisor subsystem=qemu
```

shim configuration:
```sh
[hypervisor.qemu]
path = "/opt/kata/bin/qemu-system-x86_64"
kernel = "/opt/kata/share/kata-containers/vmlinux.container"
initrd = "/opt/kata/share/kata-containers/kata-containers-initrd.img"
machine_type = "q35"
valid_hypervisor_paths = ["/opt/kata/bin/qemu-system-x86_64"]
kernel_params = ""
firmware = ""
machine_accelerators=""
cpu_features="pmu=off"
default_vcpus = 1
default_maxvcpus = 0
default_bridges = 1
default_memory = 2048
disable_block_device_use = false
shared_fs = "virtio-fs"
virtio_fs_daemon = "/opt/kata/libexec/kata-qemu/virtiofsd"
valid_virtio_fs_daemon_paths = ["/opt/kata/libexec/kata-qemu/virtiofsd"]
virtio_fs_cache_size = 0
virtio_fs_extra_args = ["--thread-pool-size=1", "-o", "log_level=debug"]
virtio_fs_cache = "auto"
block_device_driver = "virtio-scsi"
enable_iothreads = false
enable_vhost_user_store = false
vhost_user_store_path = "/var/run/kata-containers/vhost-user"
valid_vhost_user_store_paths = ["/var/run/kata-containers/vhost-user"]
valid_file_mem_backends = [""]
enable_swap = true                                               <<--- Cause of the problem
pflashes = []
pcie_root_port = 2
valid_entropy_sources = ["/dev/urandom","/dev/random",""]
[factory]
[agent.kata]
enable_debug = true
kernel_modules=[]
debug_console_enabled = true
[netmon]
path = "/opt/kata/libexec/kata-containers/kata-netmon"
[runtime]
enable_debug = true
internetworking_model="tcfilter"
disable_guest_seccomp=true
sandbox_cgroup_only=true
sandbox_bind_mounts=[]
experimental=[]
```

qick creating pod example yaml:
```sh
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: ubuntu-kata
  labels:
    app: ubuntu-kata
spec:
  runtimeClassName: kata
  containers:
  - name: ubuntu
    image: ubuntu:20.04
    resources:
      limits:
        cpu: 2
        memory: 1Gi  
EOF
```

**Comment by Bevisy** [comment]
# quick look
error log:
```sh
"failed to launch qemu: qemu-system-x86_64: -realtime: invalid option\n"
```
error option:
```sh
-realtime mlock=off
```

**Comment by Bevisy** [comment]
Looks like I found the root cause:
```sh
-realtime (removed in 6.0)
The -realtime mlock=on|off argument has been replaced by the -overcommit mem-lock=on|off argument.
```
refer [removed-features](https://qemu.readthedocs.io/en/stable/about/removed-features.html)

**Comment by liubin** [comment]
Related https://github.com/kata-containers/kata-containers/issues/3080

**Comment by Bevisy** [comment]
Releated https://github.com/kata-containers/govmm/pull/190

Hold on until #3081 merged

**Comment by Bevisy** [comment]
closed by 49223e67af3b2990d32add91aea1df47148c11bd
